src/libraries/System.Private.CoreLib/src/System/AggregateException.cs

```diff
@@ -118,43 +116,29 @@ public AggregateException(string? message, IEnumerable<Exception> innerException
         /// <exception cref="System.ArgumentException">An element of <paramref name="innerExceptions"/> is
         /// null.</exception>
         public AggregateException(string? message, params Exception[] innerExceptions) :
-            this(message, (IList<Exception>)innerExceptions)
+            this(message, innerExceptions, cloneExceptions: true)
         {
         }
 
-        /// <summary>
-        /// Allocates a new aggregate exception with the specified message and list of inner exceptions.
-        /// </summary>
-        /// <param name="message">The error message that explains the reason for the exception.</param>
-        /// <param name="innerExceptions">The exceptions that are the cause of the current exception.</param>
-        /// <exception cref="System.ArgumentNullException">The <paramref name="innerExceptions"/> argument
-        /// is null.</exception>
-        /// <exception cref="System.ArgumentException">An element of <paramref name="innerExceptions"/> is
-        /// null.</exception>
-        private AggregateException(string? message, IList<Exception> innerExceptions)
-            : base(message, innerExceptions != null && innerExceptions.Count > 0 ? innerExceptions[0] : null)
+        private AggregateException(string? message, Exception[] innerExceptions, bool cloneExceptions) :
+            base(message, innerExceptions?.Length > 0 ? innerExceptions[0] : null)
         {
             if (innerExceptions == null)
             {
                 throw new ArgumentNullException(nameof(innerExceptions));
             }
 
-            // Copy exceptions to our internal array and validate them. We must copy them,
-            // because we're going to put them into a ReadOnlyCollection which simply reuses
-            // the list passed in to it. We don't want callers subsequently mutating.
-            Exception[] exceptionsCopy = new Exception[innerExceptions.Count];
+            _innerExceptions = cloneExceptions ? new Exception[innerExceptions.Length] : innerExceptions;
```

Comment:
`(Exception[])innerExceptions.Clone()`?

Reply:
Is it better with the cast from object?

Reply:
Sorry, not sure what you mean by that question.

Reply:
I meant is Clone version better if it has to generate a cast check?